context/app/static/js/components/Routes/Routes.jsx

```diff
@@ -11,13 +11,16 @@ import Preview from '../Preview';
 import { Collections } from '../Collections';
 import Markdown from '../Markdown';
 import useSendPageView from './useSendPageView';
+import useSetUrlPathAndSearchBeforeLogin from './useSetUrlPathAndSearchBeforeLogin';
 
 function Routes(props) {
   const { flaskData } = props;
   const { entity, vitessce_conf, title, markdown, collection, errorCode } = flaskData;
   const urlPath = window.location.pathname;
+  const pathAndSearch = (window.location.pathname + window.location.search).substr(1);
```

Comment:
Why not capture the whole URL?

Reply:
Yeah, and then the `external=True` becomes moot. And if there's a fragment in the URL, good to capture it, too.